com.reprezen.swagedit/src/com/reprezen/swagedit/json/references/JsonReferenceValidator.java

```diff
@@ -60,27 +60,25 @@ public JsonReferenceValidator(JsonReferenceFactory factory) {
 	protected Collection<? extends SwaggerError> doValidate(URI baseURI, Iterable<JsonReference> references) {
 		Set<SwaggerError> errors = Sets.newHashSet();
 		for (JsonReference reference: references) {
-			if (!reference.isValid(baseURI)) {
-				Object source = reference.getSource();
-
-				if (source instanceof Node) {
-					errors.add(createReferenceError((Node) source));
-				} else {
-					errors.add(createReferenceError());
-				}
+			if (reference.isInvalid()) {
+				errors.add(createReferenceError(IMarker.SEVERITY_ERROR, reference));
+			} else if (reference.isMissing(baseURI)) {
+				errors.add(createReferenceError(IMarker.SEVERITY_WARNING, reference));
 			}
 		}
 		return errors;
 	}
 
-	protected SwaggerError createReferenceError(Node node) {
-		int line = node.getStartMark().getLine() + 1;
-
-		return new SwaggerError(line, IMarker.SEVERITY_WARNING, Messages.error_invalid_reference);
-	}
+	protected SwaggerError createReferenceError(int severity, JsonReference reference) {
+		Object source = reference.getSource();
+		int line;
+		if (source instanceof Node) {
+			line = ((Node) source).getStartMark().getLine() + 1;
+		} else {
+			line = 1;
+		}
 
-	protected SwaggerError createReferenceError() {
-		return new SwaggerError(1, IMarker.SEVERITY_WARNING, Messages.error_invalid_reference);
+		return new SwaggerError(line, severity, Messages.error_invalid_reference);
```

Comment:
It would be better to change the first line to this:

> Invalid Reference Syntax - The referenced path or URI may contain invalid characters.

The second line can remain the same.